src/function/table/range/int_range_executor.cpp

```diff
@@ -0,0 +1,117 @@
+#include "duckdb/function/table/range/int_range_executor.hpp"
+
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/common/algorithm.hpp"
+#include "duckdb/execution/expression_executor.hpp"
+#include "duckdb/planner/expression/bound_reference_expression.hpp"
+
+namespace duckdb {
+
+//===--------------------------------------------------------------------===//
+// Range (timestamp)
+//===--------------------------------------------------------------------===//
+
+void IntRangeExecutor::Reset() {
+	RangeExecutor::Reset();
+	current_row = 0;
+}
+
+void IntRangeExecutor::FetchArguments(DataChunk &input, const vector<unique_ptr<Expression>> &args_list) {
+
+	args_data.Reset();
+	// execute the argument expressions
+	// execution results (arguments) are kept in args_data chunk
+	expr_executor.Execute(input, args_data);
+
+	// verify incoming arguments
+	args_data.Verify();
+	D_ASSERT(input.size() == args_data.size());
+	D_ASSERT(args_data.ColumnCount() == args_list.size());
+	D_ASSERT(args_list.size() <= 3 && args_list.size() >= 1);
+
+	first_fetch = false;
+
+	vector<Value> args_values {};
+
+	for (idx_t i = 0; i < args_list.size(); i++) {
+		Value v = args_data.GetValue(i, 0);
```

Comment:
I'm quite sure this is a problem, because we're not considering the amount of rows of the input chunk.
Because `range/generate_series` is now a table in-out function that means we can get a table as input, meaning we can have multiple input rows.

This always takes the first row.

Reply:
Thanks for the comment!
I debugged the test query `SELECT COUNT(*) FROM range(0,10) AS _(r1), range(r1,10) __(r2);` to see whether or not we're actually dealing with multiple input rows when `FetchArguments` is called (during evaluation of `range(r1,10)`). Turns out the handling of the table input has already been done previously (ll. 65-100 in `physical_tableinout_function.cpp`) and all involved chunks only have a single row at this stage:
- `input` has one single row (containing the correct value for `r1` in each iteration)
- l. 24 executes the bound reference expressions and stores them in the `args_data` chunk, which therefore also has only a single row

Therefore I assume using the first row to be safe. If you agree, I'll simply add an appropriate assertion.